api/src/wfl/module/covid.clj

```diff
@@ -422,6 +422,39 @@
   (cond (= "importSnapshot" fromSource) (import-snapshot! executor source-item)
         :else (throw (ex-info "Unknown fromSource" {:executor executor}))))
 
+(defn ^:private update-method-configuration
+  "Update METHODCONFIGURATION in WORKSPACE with snapshot reference NAME
+  as :dataReferenceName via Firecloud.
+  Update executor table record ID with incremented METHODCONFIGURATIONVERSION.
+  Return EXECUTOR with incremented METHODCONFIGURATIONVERSION."
+  [{:keys [id
+           workspace
+           methodConfiguration
+           methodConfigurationVersion] :as executor}
+   {:keys [name]                       :as _reference}]
+  (let [{:keys [methodConfigVersion] :as firecloud-mc}
+        (try
+          (firecloud/get-method-configuration workspace methodConfiguration)
+          (catch Throwable t
+            (throw (ex-info "Method configuration does not exist in workspace"
+                            {:executor executor
+                             :cause    (.getMessage t)}))))
+        inc-version (inc methodConfigurationVersion)]
+    (when-not (== methodConfigurationVersion methodConfigVersion)
+      (throw (ex-info (str "Firecloud method configuration version != expected: "
+                           "possible concurrent modification")
+                      {:executor            executor
+                       :methodConfiguration firecloud-mc})))
+    (-> firecloud-mc
```

Comment:
Pretty much copy-pasted from the [COVID demo script](https://github.com/broadinstitute/wfl/blob/develop/api/test/wfl/system/cdc_covid19_surveillance_demo.clj#L76).